src/main/java/com/rog/EShop/repository/ItemRepository.java

```diff
@@ -1,14 +1,18 @@
 package com.rog.EShop.repository;
 
+import com.rog.EShop.dto.StatsDto;
 import com.rog.EShop.entity.Item;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 import java.util.List;
 
 public interface ItemRepository extends JpaRepository<Item, Integer> {
-    Page<Item> findFirst5ByOrderByIdDesc(Pageable pageable);
     Page<Item> findAllByCategoryId(Integer id, Pageable pageable);
+    @Query("select new com.rog.EShop.dto.StatsDto(c.name, count(i.id)) from Item i " +
```

Comment:
there are some problems in this request:
1. Query is not working. `JOIN Catergory c` is wrong, in JPA you need to write `JOIN i.category c`
2. If category has no items, it not shown in response. Need to show all categories, even empty.
3. You could use right join, but in this case it is impossible, because RIGHT JOIN in not supported in JPA.